library/include/rocblas_bfloat16.h

```diff
@@ -74,45 +74,58 @@ struct rocblas_bfloat16
     }
 
 private:
-    static constexpr __host__ __device__ uint16_t float_to_bfloat16(float f)
+    static constexpr __host__ __device__ uint16_t float_to_bfloat16(float f, bool round = true)
```

Comment:
It would be better if this truncation function were implemented somewhere in `clients/*` instead of in `library/include/rocblas_bfloat16.h`. Do you need it anywhere outside of the tests?

There is an associated `tensile_bfloat16.h`. Right now only the names, as the use of C++14 `constexpr`, is different between `rocblas_bfloat16.h` and `tensile_bfloat16.h`. This change would be making them diverge further.

Reply:
Aha, didn't notice we had a tensile_bfloat16 as well, I can add this change to that too. As a side note, would it make sense to have a common shared library for things like this? The only difference looks to be the name (tensile_bfloat16 vs rocblas_bfloat16).

Reply:
I still don't think we should be messing with `rocblas_bfloat16.h` (or `tensile_bfloat16.h`).

This truncation instead of rounding seems to be a corner case for specific kernels -- why are they truncating instead of rounding properly? Since it's a corner case for specific kernels and we just need to test them, the truncation function belongs in the tests and not in the global, public header files `rocblas_bfloat16.h` and `tensile_bfloat16.h`.